allennlp/common/params.py

```diff
@@ -188,15 +188,21 @@ def _check_is_dict(self, new_history, value):
         return value
 
     @staticmethod
-    def from_file(params_file: str) -> 'Params':
+    def from_file(params_file: str, params_overrides: str = None) -> 'Params':
         """
         Load a `Params` object from a configuration file.
         """
         # redirect to cache, if necessary
         params_file = cached_path(params_file)
 
-        param_dict = pyhocon.ConfigFactory.parse_file(params_file)
-        return Params(param_dict)
+        file_dict = pyhocon.ConfigFactory.parse_file(params_file)
+
+        if params_overrides != None:
```

Comment:
idiomatic is to use `is not None`

Reply:
but I feel like this method could be much cleaner if the default value for `params_overrides` was the empty string.

then you don't need a conditional at all, you can just do

```
original = pyhocon.ConfigFactory.parse_file(params_file)
overrides = pyhocon.ConfigFactory.parse_string(params_overrides)
return Params(overrides.with_fallback(original))
```

much simpler

Reply:
Funny--you're right.  I'm misremembering `pyhocon.ConfigFactory.parse_string("")` throwing an exception.